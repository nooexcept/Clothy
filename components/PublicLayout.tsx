import React from 'react'
import clsx from 'clsx'
import { makeStyles, Theme, createStyles } from '@material-ui/core/styles'
import Box from '@material-ui/core/Box'
import PublicDrawer from './PublicDrawer'
import PublicAppBar from './PublicAppBar'

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            display: 'flex',
        },
        grow: {
            flexGrow: 1,
        },
        content: {
            flexGrow: 1,
            padding: theme.spacing(3),
            transition: theme.transitions.create('margin', {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.leavingScreen,
            }),
            marginLeft: -240,
        },
        contentShift: {
            transition: theme.transitions.create('margin', {
                easing: theme.transitions.easing.easeOut,
                duration: theme.transitions.duration.enteringScreen,
            }),
            marginLeft: 0,
        },
    })
)

const PublicLayout: React.FC = ({
    children,
}: {
    children: React.ReactChildren
}) => {
    const classes = useStyles()
    const [open, setOpen] = React.useState(false)

    return (
        <React.Fragment>
            <div className={classes.root}>
                <PublicAppBar setOpen={setOpen} open={open} />
                <PublicDrawer setOpen={setOpen} open={open} />
                <main
                    className={clsx(classes.content, {
                        [classes.contentShift]: open,
                    })}
                >
                    <Box paddingTop={8}>{children}</Box>
                </main>
            </div>
        </React.Fragment>
    )
}

export default PublicLayout
