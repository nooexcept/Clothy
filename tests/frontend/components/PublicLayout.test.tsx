import React from 'react'
import { ShallowWrapper } from 'enzyme'
import { createShallow } from '@material-ui/core/test-utils'
import PublicLayout from '../../../components/PublicLayout'

let shallow: ReturnType<typeof createShallow>
let wrapper: ShallowWrapper

beforeEach(() => {
    shallow = createShallow()
    wrapper = shallow(
        <PublicLayout>
            <p>Hello world</p>
        </PublicLayout>
    )
})

describe('<PublicLayout /> rendering', () => {
    it('should render without problems', () => {
        expect(wrapper).toMatchSnapshot()
    })
})
