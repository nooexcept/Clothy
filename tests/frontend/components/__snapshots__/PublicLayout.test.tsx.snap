// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PublicLayout /> rendering should render without problems 1`] = `
<Fragment>
  <div
    className="makeStyles-root-1"
  >
    <PublicAppBar
      open={false}
      setOpen={[Function]}
    />
    <PublicDrawer
      open={false}
      setOpen={[Function]}
    />
    <main
      className="makeStyles-content-3"
    >
      <Styled(MuiBox)
        className="makeStyles-root-1"
        paddingTop={8}
      >
        <p>
          Hello world
        </p>
      </Styled(MuiBox)>
    </main>
  </div>
</Fragment>
`;
